delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/GpuDeltaTaskStatisticsTracker.scala

```diff
@@ -152,11 +154,13 @@ class GpuDeltaTaskStatisticsTracker(
     })
   }
 
-  override def newBatch(filePath: String, batch: ColumnarBatch): Unit = {
+  override def newBatch(filePath: String, spillableBatch: SpillableColumnarBatch): Unit = {
     val aggBuffer = submittedFiles(filePath)
     extendedRow.update(0, aggBuffer)
 
-    batchStatsToRow(batch, gpuResultsBuffer)
+    withResource(spillableBatch.getColumnarBatch()) { batch =>
```

Comment:
Should we have a follow on issue so we can put this in a retry instead of just getting the columnar batch?  It looks like `batchStatsToRow` has a side effect so there is no good way to do it here.

Reply:
Why not do:

```
val batch = withRetryNoSplit {
  spillableBatch.getColumnarBatch()
}

withResource(batch) { batchStatsToRow(_, gpuResultsBuffer) } 
```

Reply:
Oh I see, `batchStatsToRow` is also doing GPU work, so it could fail.

Reply:
Added https://github.com/NVIDIA/spark-rapids/issues/8670 for follow on and will do the `withRetryNoSplit` as mentioned above. Added comment in the code.